lib/src/dsl_util.rs

```diff
@@ -429,6 +430,28 @@ impl<R: RuleType> StringLiteralParser<R> {
     }
 }
 
+/// Escape special characters in the input
+pub fn escape_string(unescaped: &str) -> String {
+    let mut escaped = String::with_capacity(unescaped.len());
+    for c in unescaped.chars() {
+        match c {
+            '"' => escaped.push_str(r#"\""#),
+            '\\' => escaped.push_str(r#"\\"#),
+            '\t' => escaped.push_str(r#"\t"#),
+            '\r' => escaped.push_str(r#"\r"#),
+            '\n' => escaped.push_str(r#"\n"#),
+            '\0' => escaped.push_str(r#"\0"#),
+            c if c < '\x20' || c == '\x7f' => {
```

Comment:
nit: `.is_ascii_contro()` or range pattern can be used

Reply:
Ah, nice. Switched to `.is_ascii_control()`